src/sdk/propagation/trace_context.rs

```diff
@@ -102,42 +102,42 @@ impl TraceContextPropagator {
                 .unwrap_or_else(|_| TraceState::default());
 
         // create context
-        let span_context = SpanContext::new(trace_id, span_id, trace_flags, true, trace_state);
+        let span_reference = SpanReference::new(trace_id, span_id, trace_flags, true, trace_state);
 
         // Ensure span is valid
-        if !span_context.is_valid() {
+        if !span_reference.is_valid() {
             return Err(());
         }
 
-        Ok(span_context)
+        Ok(span_reference)
     }
 }
 
 impl TextMapPropagator for TraceContextPropagator {
-    /// Properly encodes the values of the `SpanContext` and injects them
+    /// Properly encodes the values of the `SpanReference` and injects them
     /// into the `Injector`.
     fn inject_context(&self, cx: &Context, injector: &mut dyn Injector) {
-        let span_context = cx.span().span_context();
-        if span_context.is_valid() {
+        let span_reference = cx.span().span_reference();
+        if span_reference.is_valid() {
             let header_value = format!(
                 "{:02x}-{:032x}-{:016x}-{:02x}",
                 SUPPORTED_VERSION,
-                span_context.trace_id().to_u128(),
-                span_context.span_id().to_u64(),
-                span_context.trace_flags() & TRACE_FLAG_SAMPLED
+                span_reference.trace_id().to_u128(),
+                span_reference.span_id().to_u64(),
+                span_reference.trace_flags() & TRACE_FLAG_SAMPLED
             );
             injector.set(TRACEPARENT_HEADER, header_value);
-            injector.set(TRACESTATE_HEADER, span_context.trace_state().header());
+            injector.set(TRACESTATE_HEADER, span_reference.trace_state().header());
         }
     }
 
-    /// Retrieves encoded `SpanContext`s using the `Extractor`. It decodes
-    /// the `SpanContext` and returns it. If no `SpanContext` was retrieved
-    /// OR if the retrieved SpanContext is invalid then an empty `SpanContext`
+    /// Retrieves encoded `SpanReference`s using the `Extractor`. It decodes
+    /// the `SpanReference` and returns it. If no `SpanReference` was retrieved
+    /// OR if the retrieved SpanReference is invalid then an empty `SpanReference`
     /// is returned.
     fn extract_with_context(&self, cx: &Context, extractor: &dyn Extractor) -> Context {
-        self.extract_span_context(extractor)
-            .map(|sc| cx.with_remote_span_context(sc))
+        self.extract_span_reference(extractor)
+            .map(|sc| cx.with_remote_span_reference(sc))
```

Comment:
Small nit: this should be `sr` now instead of `sc`

Reply:
good catch!